roles/servicetelemetry/tasks/component_grafana.yml

```diff
@@ -47,94 +26,247 @@
-          metadata:
-            labels:
-              app: grafana
-              stf_owner: "{{ ansible_operator_meta.name }}"
-            name: virtual-machine-dashboard-1
-            namespace: "{{ ansible_operator_meta.namespace }}"
-          spec:
-            name: virtual-machine-view.json
-            json: |
-              {{ lookup('file', 'virtual-machine-view.json') | string }}
+#---- deploy Grafana with v5 Operator (preferred)
+    - when: has_grafana_integreatly_api
+      name: Deploying with Grafana Operator v5
+      block:
+        - name: Construct oauth redirect reference
+          set_fact:
+            grafana_oauth_redir_ref:
+              kind: OAuthRedirectReference
+              apiVersion: v1
+              reference:
+                kind: Route
+                name: '{{ ansible_operator_meta.name }}-grafana-route'
+
+        - name: Lookup template
+          debug:
+            msg: "{{ lookup('template', './manifest_grafana_v5.j2') | from_yaml }}"
+
+        - name: Set default Grafana manifest (Grafana Operator v5)
+          set_fact:
+            grafana_manifest: "{{ lookup('template', './manifest_grafana_v5.j2') | from_yaml }}"
+          when: grafana_manifest is not defined
+
+        - name: Create an instance of Grafana (Grafana Operator v5)
+          k8s:
+            state: '{{ "present" if servicetelemetry_vars.graphing.enabled else "absent" }}'
+            definition:
+              '{{ grafana_manifest }}'
+
+        # NOTE: we only provide events forwarding with STF. We don't use events
+        # in dashboards, so there is no need to create an Elasticsearch
+        # datasource.
+        - when: servicetelemetry_vars.backends.metrics.prometheus.enabled
+          name: Create Grafana datasource for Prometheus
+          block:
+            - name: Set datasource for Prometheus
+              set_fact:
+                ds_manifest: "{{ lookup('template', './manifest_grafana_ds_prometheus.j2') | from_yaml }}"
+              when: ds_manifest is not defined
+
+            - name: Create the datasource for Prometheus
+              k8s:
+                state: '{{ "present" if servicetelemetry_vars.graphing.enabled else "absent" }}'
+                definition:
+                  '{{ ds_manifest }}'
+
+        - name: Load Cloud Overview Dashboard
+          k8s:
+            state: '{{ "present" if servicetelemetry_vars.graphing.grafana.dashboards.enabled else "absent" }}'
+            definition:
+              apiVersion: grafana.integreatly.org/v1beta1
+              kind: GrafanaDashboard
+              metadata:
+                labels:
+                  app: grafana
+                  stf_owner: "{{ ansible_operator_meta.name }}"
+                name: rhos-cloud-dashboard-1
+                namespace: "{{ ansible_operator_meta.namespace }}"
+              spec:
+                instanceSelector:
+                  matchLabels:
+                    dashboards: "stf"
```

Comment:
You'll see the usage of `instanceSelector` everywhere because that's how association works with Grafana Operator v5.

The Grafana instance will have a traditional k8s label assigned to it as `dashboards: stf` and then the instance selectors on the GrafanaDashboard and GrafanaDatasource objects get bound to instances of Grafana with matching instance selector values.